ui/components/multichain/account-details/account-details.js

```diff
@@ -0,0 +1,203 @@
+  const exportPrivateKeyFeatureEnabled = !isHardwareKeyring(keyring?.type);
+
+  const [attemptingExport, setAttemptingExport] = useState(false);
+  const [password, setPassword] = useState('');
+
+  // Password error would result from appState
+  const warning = useSelector((state) => state.appState.warning);
+
+  // This is only populated when the user properly authenticates
+  const privateKey = useSelector(
+    (state) => state.appState.accountDetail.privateKey,
+  );
+
+  const onClose = useCallback(() => {
+    dispatch(setAccountDetailsAddress(''));
+    dispatch(clearAccountDetails());
+  }, [dispatch]);
+
+  const onSubmit = useCallback(() => {
+    dispatch(exportAccount(password, address)).then((res) => {
+      dispatch(hideWarning());
+      return res;
+    });
+  }, [dispatch, password, address]);
+
+  return (
+    <Popover
+      headerProps={{
+        paddingBottom: 1,
+      }}
+      contentProps={{
+        paddingLeft: 4,
+        paddingRight: 4,
+        paddingBottom: 4,
+      }}
+      title={
+        <PopoverHeader
+          childrenWrapperProps={{
+            display: DISPLAY.FLEX,
+            justifyContent: JustifyContent.center,
+          }}
+        >
+          <AvatarAccount
+            variant={
+              useBlockie
+                ? AvatarAccountVariant.Blockies
+                : AvatarAccountVariant.Jazzicon
+            }
+            address={address}
+            size={AvatarAccountSize.Lg}
+          />
+        </PopoverHeader>
+      }
+      onClose={onClose}
+    >
+      <Box
+        display={DISPLAY.FLEX}
+        alignItems={AlignItems.center}
+        flexDirection={FLEX_DIRECTION.COLUMN}
+      >
+        {attemptingExport === false ? (
+          <>
+            <EditableLabel
+              defaultValue={name}
+              onSubmit={(label) => dispatch(setAccountLabel(address, label))}
+              accounts={accounts}
+            />
+            <QrView Qr={{ data: address }} />
+            {exportPrivateKeyFeatureEnabled ? (
+              <ButtonSecondary
+                block
+                size={BUTTON_SECONDARY_SIZES.LG}
+                variant={TextVariant.bodyMd}
+                onClick={() => {
+                  trackEvent({
+                    category: MetaMetricsEventCategory.Accounts,
+                    event: MetaMetricsEventName.KeyExportSelected,
+                    properties: {
+                      key_type: MetaMetricsEventKeyType.Pkey,
+                      location: 'Account Details Modal',
+                    },
+                  });
+                  /*
+                  dispatch(showModal({ name: 'EXPORT_PRIVATE_KEY' }));
+                  onClose();
+                  */
+                  setAttemptingExport(true);
+                }}
+              >
+                {t('showPrivateKey')}
+              </ButtonSecondary>
+            ) : null}
+          </>
+        ) : null}
```

Comment:
We can move this logic inside same ternary operator below
```suggestion

```

Reply:
Good shout!  I was moving fast, just didn't see it.